libsolidity/parsing/Parser.cpp

```diff
@@ -2076,16 +2078,39 @@ optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> cons
 	for (auto const& [start, end]: sequencesToSearch)
 	{
 		smatch match;
-		if (regex_search(start, end, match, licenseRegex))
+		auto words_begin =
+			std::sregex_iterator(start, end, licenseRegex);
+		auto words_end = std::sregex_iterator();
+
+		for (std::sregex_iterator i = words_begin; i != words_end; ++i)
```

Comment:
The naming bothers me a bit here. I was initially a bit confused about why you match words if the regex is for the whole declaration :) Also, it does not follow our camel case convention.

```suggestion
		auto declarationsBegin = std::sregex_iterator(start, end, licenseRegex);
		auto declarationsEnd = std::sregex_iterator();

		for (std::sregex_iterator declarationIt = declarationsBegin; declarationIt != declarationsEnd; ++declarationIt)
```